examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,224 @@
+                        }
+                        if constexpr (std::is_same_v<T, sf::Event::MouseButtonPressed>)
+                        {
+                            m_log.emplace_back("2: Mouse Pressed");
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::TouchBegan>)
+                        {
+                            m_log.emplace_back("2: Touch Began: " + vec2ToString(event.position));
+                        }
+                        else
+                        {
+                            // All unhandled events will end up here
+                            // m_log.emplace_back("2: Other Event");
+                        }
+                    });
+            }
+            else if (handlerType == 3)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Switch to next form of event handling
+            handlerType = (handlerType + 1) % 4;
+
+            while (m_log.size() > 24u)
+                m_log.erase(m_log.begin());
+
+            // Draw the contents of the log to the window
+            m_window.clear();
+
+            for (std::size_t i = 0; i < m_log.size(); ++i)
+            {
+                m_logText.setPosition({50.f, static_cast<float>(i * 20) + 50.f});
+                m_logText.setString(m_log[i]);
+                m_window.draw(m_logText);
+            }
+
+            m_window.display();
+        }
+    }
+
+    void handle(const sf::Event::Closed&)
+    {
+        m_window.close();
+    }
+
+    void handle(const sf::Event::KeyPressed& keyPress)
+    {
+        m_log.emplace_back("3: Key Pressed: " + sf::Keyboard::getDescription(keyPress.scancode));
+    }
+
+    void handle(const sf::Event::MouseMoved& mouseMoved)
+    {
+        m_log.emplace_back("3: Mouse Moved: " + vec2ToString(mouseMoved.position));
+    }
+
+    void handle(const sf::Event::MouseButtonPressed&)
+    {
+        m_log.emplace_back("3: Mouse Pressed");
+    }
+
+    void handle(const sf::Event::TouchBegan& touchBegan)
+    {
+        m_log.emplace_back("3: Touch Began: " + vec2ToString(touchBegan.position));
+    }
+
+    template <typename T>
+    void handle(const T&)
+    {
+        // All unhandled events will end up here
+        // m_log.emplace_back("3: Other Event");
+    }
```

Comment:
This block should have its own comment explaining how it's being used. Currently this feels a bit out of the blue, if you haven't read the specific case's details/implementation above.

Reply:
Fixed.